### PR TITLE
Unused imports validation

### PR DESCRIPTION
**Pull request by nsk-mironov**
Still Work In Progress.

### CODE REVIEW

File: lightsaber/gradle-plugin/src/main/java/com/joom/lightsaber/plugin/JavaLightsaberPluginExtension.kt [hunk 1]
```diff
@@ -19,5 +19,6 @@ package com.joom.lightsaber.plugin
 open class JavaLightsaberPluginExtension {
   var processTest: Boolean = true
   var validateUsage: Boolean = true
+  var validateUnusedImports: Boolean = false
```
Comment (MichaelRocks) [inline lightsaber/gradle-plugin/src/main/java/com/joom/lightsaber/plugin/JavaLightsaberPluginExtension.kt:22]: Why isn't it `true` by default like `validateUsage`?

File: lightsaber/processor/src/main/java/com/joom/lightsaber/processor/validation/DependencyResolver.kt [hunk 2]
```diff
@@ -181,6 +182,15 @@ class DependencyResolverImpl(
...
     }
+    if (includeAllDependenciesInGraph) {
+      for (factoryProvisionPoint in factory.provisionPoints) {
+        for (injectee in factoryProvisionPoint.injectionPoint.injectees) {
+          if (injectee is FactoryInjectee.FromInjector) {
+            dependencyGraph.put(factory.dependency, injectee.dependency)
+          }
+        }
+      }
+    }
```
Comment (MichaelRocks) [inline lightsaber/processor/src/main/java/com/joom/lightsaber/processor/validation/DependencyResolver.kt]: Maybe this code should be moved to the `add` method called above?

**Review by MichaelRocks — APPROVED** [review]

File: lightsaber/processor/src/main/java/com/joom/lightsaber/processor/validation/DependencyResolver.kt (hunk 2 above)
Comment (nsk-mironov) [reply lightsaber/processor/src/main/java/com/joom/lightsaber/processor/validation/DependencyResolver.kt]: Fixed

File: lightsaber/gradle-plugin/src/main/java/com/joom/lightsaber/plugin/JavaLightsaberPluginExtension.kt (hunk 1 above)
Comment (nsk-mironov) [reply lightsaber/gradle-plugin/src/main/java/com/joom/lightsaber/plugin/JavaLightsaberPluginExtension.kt:22]: I'm planning on making a separate PR that will:
- Enable `validateUnusedImports` by default
- Make it possible to change the default value for all modules via single `gradle.properties` property
- Add an explanation to an error message on how to disable the validation

Enabling `validateUnusedImports` by default will break existing projects and currently it's really hard to disable it in a multi-module environment, that's why I want it to be disabled by default for now.

**Pull request merged**